charts/opentelemetry-collector/templates/daemonset.yaml

```diff
@@ -9,15 +9,13 @@ spec:
   selector:
     matchLabels:
       {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
-      component: agent-collector
```

Comment:
This change breaks the upgrade process. Users will have to reinstall the chart.

This has to be well documented. Also please provide more justification for this change with references.

And it would be nice if you can find an example of graceful handling of this change in other helm charts.